### PR TITLE
Update docstring of transforms that can be applied to `QNode` and state wrong return type.

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
@timmysilv noticed that the "Returns" statements of gradient transforms are off: They claim that such a transform, e.g. `param_shift`, returns a `tensor_like`, `tuple[tensor_like]` or `tuple[tuple[tensor_like]]` when applied to a `QNode`. However, the transform returns a `callable`, which in turn has the return type currently described by the docstring.

**Description of the Change:**
The docstring of the following functions is changed to make the return type correct and (hopefully) make it even more clearer what is the difference between calling a transform on a tape or on a QNode.

- `gradients.finite_diff`,
- `gradients.param_shift`,
- `gradients.param_shift_cv`,
- `gradients.param_shift_hessian`,
- internal functions that are called by these four functions.
- `optimize.lie_algebra.algebra_commutator`
- `transforms.broadcast_expand`
- `transforms.metric_tensor`
- `transforms.mitigate.fold_global`

Some other tiny changes have been made to other docstrings to clarify details.

**Benefits:**
Documentation is improved/corrected.

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
Not an issue, but the observation that this is off, can be found [here](https://github.com/PennyLaneAI/pennylane/pull/3366#discussion_r1038428270)

### CODE REVIEW

**Review by rmoyard — APPROVED** [review]
Thank you for clarifying the documentation @dwierichs, it looks good to me!

File: pennylane/gradients/parameter_shift_cv.py [hunk 1]
```diff
@@ -522,15 +524,16 @@ def param_shift_cv(
         force_order2 (bool): if True, use the order-2 method even if not necessary
 
     Returns:
-        tensor_like or tuple[list[QuantumTape], function]:
+        function or tuple[list[QuantumTape], function]:
 
-        - If the input is a QNode, a tensor
-          representing the output Jacobian matrix of size ``(number_outputs, number_gate_parameters)``
-          is returned.
+        - If the input is a QNode, an object representing the Jacobian  (function)of the QNode
```
Comment (timmysilv) [inline pennylane/gradients/parameter_shift_cv.py:529]:
```suggestion
        - If the input is a QNode, an object representing the Jacobian (function) of the QNode

```

File: pennylane/gradients/parameter_shift_hessian.py [hunk 2]
```diff
@@ -740,17 +744,19 @@ def _param_shift_hessian_tuple(
...
     Returns:
-        tensor_like or tuple[tensor_like] or tuple[list[QuantumTape], function]:
-
-        - If the input is a QNode with a single trainable argument, a tensor representing the
-          Hessian of size ``(*QNode output dimensions, *QNode input dimensions, *QNode input dimensions)``
-          is returned.
+        function or tuple[list[QuantumTape], function]:
 
-        - If the input is a QNode with multiple trainable arguments, a tuple of Hessian tensors is
+        - # TODO: If the input is a QNode, an object representing the Hessian (function) of
```
Comment (timmysilv) [inline pennylane/gradients/parameter_shift_hessian.py:749]: what is the TODO here?

Comment (dwierichs) [reply pennylane/gradients/parameter_shift_hessian.py:749]: Ah, right! I am not entirely sure the statement is true :sweat_smile: 
I also found a bug while trying to find out, which I have to put up still :)

Comment (dwierichs) [reply pennylane/gradients/parameter_shift_hessian.py]: Fixed this :)

**Pull request merged**